service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateAzureIpStep.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.workspace.service.resource.controlled.flight.create;
+
+import bio.terra.cloudres.azure.resourcemanager.common.Defaults;
+import bio.terra.cloudres.azure.resourcemanager.compute.data.CreatePublicIpRequestData;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledAzureIpResource;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.Region;
+import com.azure.resourcemanager.compute.ComputeManager;
+import com.azure.resourcemanager.network.models.IpAllocationMethod;
+
+/** Creates an Azure IP address. Designed to run directly after {@link GetAzureIpStep}. */
+public class CreateAzureIpStep implements Step {
+
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureIpResource resource;
+  private final AzureCloudContext azureCloudContext;
+
+  public CreateAzureIpStep(
+      AzureConfiguration azureConfig,
+      AzureCloudContext azureCloudContext,
+      CrlService crlService,
+      ControlledAzureIpResource resource) {
+    this.azureConfig = azureConfig;
+    this.azureCloudContext = azureCloudContext;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+
+    computeManager
```

Comment:
I think you still want the code here to tolerate the IP existing. On a failure after the create and before step completion, we might restart at this step and re-issue the create.